### ISSUE TITLE
Computer freezes entirely when running  blech_clust_run() and detect_spikes()

### ISSUE THREAD

**Issue by danielsvedberg** [nubs01/blechpy#4]
Using the example workflow in docs, for some reason the entire computer just freezes when running blech_clust_run().

**Comment by nubs01** [comment]
Try updating blech clust. I was working with buzar on this last week.
Should be version 1.1.13 I think

**Comment by nubs01** [comment]
Run it with multi_process=False

**Comment by danielsvedberg** [comment]
[After reading this blog post](https://codewithoutrules.com/2018/09/04/python-multiprocessing/), I believe this problem may be related to issues with multiprocessing.Pool() fork(), and could be remedied by changing line 776 of datastructures/dataset.py, in the exp_clust branch, to "spawn" processes instead of fork()-ing them. I believe the line could just be re-written as:

`pool = multiprocessing.get_context("spawn").Pool(n_cores)`

I'm not sure if I've got the syntax down quite right. I can try to make a branch implementing it later next week if Roshan doesn't. 

A general update on the current state for anyone in the lab who runs into this issue at some point or another:

As of today, I've tested blechpy ver. 2.0.3 on my Ubuntu machine, and Abuzar has tested a 1.x.x version on his Ubuntu machine. For both of us, but not Roshan, our computers freeze entirely when running detect_spikes() and blech_clust_run(). We've determined this is not due to memory overload. The only salient difference between our computers is that Abuzar and I have more cores (32 and 12 respectively) compared to Roshan's 6 cores. 

Roshan narrowed down the source of the bug to multiprocessing, and with 2.0.3 he pushed the current workaround, which is to run detect_spikes(multi_process=False) and blech_clust_run(multi_process=False), which I have verified works on my machine.
